lib/invisible_captcha/controller_ext.rb

```diff
@@ -17,9 +18,25 @@ def invisible_captcha(options = {})
     end
 
     private
+    
+    InvisibleCaptchaValues = Struct.new(:timestamp, :spinner_value)
+    
+    def invisible_captcha_values
+      @invisible_captcha_values ||= InvisibleCaptchaValues.new(invisible_captcha_timestamp.to_s,spinner_value)
+    end
+    
+    def invisible_captcha_timestamp
+      @invisible_captcha_timestamp ||= (session[:invisible_captcha_timestamp] || Time.zone.now)
+    end
+
+    def spinner_value
+       @spinner_value ||= InvisibleCaptcha.encode("#{invisible_captcha_timestamp}-#{request.remote_ip}")
+    end
 
     def detect_spam(options = {})
-      if timestamp_spam?(options)
+      if ip_spam?(options)
+        on_spam(options)
+      elsif timestamp_spam?(options)
```

Comment:
I think we should keep first the timestamp check

Reply:
changed the order.